PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -181,12 +226,65 @@ private void startBrowserSwitch(FragmentActivity activity, PayPalResponse payPal
+        } else {
+            PayPalCheckout.start(
+                    new CreateOrder() {
+                        @Override
+                        public void create(@NonNull CreateOrderActions createOrderActions) {
+                            Context appContext = activity.getApplicationContext();
+                            Uri browserSwitchUrl = browserSwitchOptions.getUrl();
+                            int requestCode = browserSwitchOptions.getRequestCode();
+                            String returnUrlScheme = browserSwitchOptions.getReturnUrlScheme();
+                            JSONObject metadata = browserSwitchOptions.getMetadata();
+                            BrowserSwitchRequest request =
+                                    new BrowserSwitchRequest(requestCode, browserSwitchUrl, metadata, returnUrlScheme);
+                            BrowserSwitchPersistentStore persistentStore = BrowserSwitchPersistentStore.getInstance();
+                            persistentStore.putActiveRequest(request, appContext);
+                            createOrderActions.set(payPalResponse.getPairingId());
+                        }
+                    },
+                    new OnApprove() {
+                        @Override
+                        public void onApprove(@NonNull Approval approval) {
+                            Intent currentIntent = activity.getIntent();
+                            String data = String.format(
+                                    "%s://onetouch/v1/success?token=%s",
+                                    braintreeClient.getReturnUrlScheme(),
+                                    approval.getData().getOrderId()
+                            );
+                            currentIntent.setData(Uri.parse(data));
+                            activity.setIntent(currentIntent);
+                        }
+                    },
+
+                    /*
+                      PayPalFlowStarted Callback is used for errors before checkout,
+                      and PayPalBrowserSwitchResultCallback doesn't apply as we wont call a browser.
+                      Implementation of onError and onCancel depend on the design of future PayPalNativeCheckout module
+                     */
+                    new OnCancel() {
+                        @Override
+                        public void onCancel() {
+                            //is this the correct analytic?
+                            braintreeClient.sendAnalyticsEvent("paypal.native.client_cancel");
+                        }
+                    },
+                    new OnError() {
+                        @Override
+                        public void onError(@NonNull ErrorInfo errorInfo) {
+                            //do we send analytic here? if so we should specify error
+                            braintreeClient.sendAnalyticsEvent("paypal.native.error");
+                        }
```

Comment:
`PayPalFlowStarted callback` is not suitable for this scenario, nor is the `BrowserSwitch` functionality as this is a native experience, so what is the correct call back that the client should use in this case? 
I'm guessing that the error handling will depend more upon the implementation of the `PayPalCheckoutNative` module. 
Also, is the analytics ok?

Reply:
Yeah we'll probably do a second pass on the analytics since we need to align them with iOS. The callback pattern is subject to change whatever is best for `PayPalNative` is what we'll go with. We have a shell module [here](https://github.com/braintree/braintree_android/tree/paypal_native) to start ironing out the interface.